### ISSUE TITLE
Interpreter: bug in the initialization of an instance variable

### ISSUE THREAD

**Issue by hutou** [crystal-lang/crystal#12763]
The interpreter does not correctly initialise an instance variable with a default value from a class property

The following code:
```crystal
module Test
  module Config
    class_property param : Int32? = 0
  end

  class Foo
    def initialize(@v = Config.param)
      p! @v
    end
  end
end

Test::Foo.new
```

with crystal run, produces
```
$ crystal run test.cr
@v # => 0
```
with crystal i, produces
```
$ crystal i test.cr
Crystal interpreter 1.6.2 (2022-11-04).
EXPERIMENTAL SOFTWARE: if you find a bug, please consider opening an issue in
https://github.com/crystal-lang/crystal/issues/new/
@v # => nil
```

**Comment by asterite** [comment]
Thanks!

Reduced:

```crystal
module Config
  @@param : Int32? = 0

  def self.param
    @@param
  end
end

p Config.param
```

It has nothing to do with instance variable initialization. It seems class parameter default values aren't set.

**Comment by asterite** [comment]
I don't know why that's failing, but something else is going on. If I run this program through the interpreter in master:

```crystal
p 1
```

I get this:

```
Crystal interpreter 1.7.0-dev [63618ff56] (2022-11-20).
EXPERIMENTAL SOFTWARE: if you find a bug, please consider opening an issue in
https://github.com/crystal-lang/crystal/issues/new/
1
Invalid memory access (signal 11) at address 0x24
[0x104f90694] *Exception::CallStack::print_backtrace:Nil +104 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x104f67790] ~procProc(Int32, Pointer(LibC::SiginfoT), Pointer(Void), Nil)@src/signal.cr:127 +320 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x1c6a6c4e4] _sigtramp +56 in /usr/lib/system/libsystem_platform.dylib
[0x105d799e0] *Crystal::Repl::Interpreter#interpret<Crystal::ASTNode+, Crystal::Type+>:Crystal::Repl::Value +464004 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x105d07d70] *Crystal::Repl::Interpreter#interpret<Crystal::ASTNode+, Hash(String, Crystal::MetaVar)>:Crystal::Repl::Value +1852 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x1062347e0] *Crystal::Repl#interpret<Crystal::Call>:Crystal::Repl::Value +112 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x10623474c] *Crystal::Repl#interpret_exit:Crystal::Repl::Value +44 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x106234950] *Crystal::Repl#run_file<String, Array(String)>:Crystal::Repl::Value +184 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x10618c3d8] *Crystal::Command#repl:(Crystal::Repl::Value | Nil) +1060 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x106187f40] *Crystal::Command#run:(Bool | Crystal::Repl::Value | Nil) +952 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x106187a28] *Crystal::Command::run<Array(String)>:(Bool | Crystal::Repl::Value | Nil) +28 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x1061879e0] *Crystal::Command::run:(Bool | Crystal::Repl::Value | Nil) +36 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x104f41590] __crystal_main +2300 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x10510233c] *Crystal::main_user_code<Int32, Pointer(Pointer(UInt8))>:Nil +12 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x1051022a4] *Crystal::main<Int32, Pointer(Pointer(UInt8))>:Int32 +60 in /Users/aryborenszweig/Projects/crystal/.build/crystal
[0x104f4a644] main +12 in /Users/aryborenszweig/Projects/crystal/.build/crystal
```

So something between 1.6.2 and master maybe broke the interpreter. I won't have time to fix this, sorry!

**Comment by hutou** [comment]
Ok, no problem, as it works on compiled app !

**Comment by HertzDevil** [comment]
Fixed by #12635
